m2cgen/ast.py

```diff
@@ -144,6 +194,15 @@ def __str__(self):
         ])
         return "BinNumExpr(" + args + ")"
 
+    def __eq__(self, other):
+        return (type(other) is BinNumExpr and
```

Comment:
This block of code looks pretty repetitive. Can we generalize it? Eg.:
```
def _eq_bin_exprs(expr_self, expr_other, expected_type):
        return (type(expr_self) is expected_type and
                expr_self.left == expr_other.left and
                expr_self.right == expr_other.right and
                expr_self.op == expr_other.op)

def __eq__(self, other):
    return _eq_bin_exprs(self, other, BinNumExpr)
```

Reply:
Refactored.